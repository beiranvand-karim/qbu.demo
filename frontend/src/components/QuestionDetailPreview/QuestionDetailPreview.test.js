/* global describe, it, jest, expect*/
import {shallow} from "enzyme"
import {QuestionDetailPreview} from "./"
import React from "react"
import renderer from "react-test-renderer"
import {MemoryRouter} from "react-router-dom"

describe('<QuestionDetailPreview />', () => {
   const question = {
      "text": "this is a question",
      "title": "this question title",
      "prize": 252,
      "options": ["option 1", "option 2", "option 3", "option 4"]
   };
   it('should render without crashing', () => {
      shallow(<QuestionDetailPreview {...question}/>)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><QuestionDetailPreview {...question}/></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });
});