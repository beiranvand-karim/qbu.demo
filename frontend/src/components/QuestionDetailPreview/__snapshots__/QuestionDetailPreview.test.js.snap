// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionDetailPreview /> should matches the snapshot 1`] = `
Array [
  <h3>
    this question title
  </h3>,
  <h4>
    252
  </h4>,
  <div>
    <img
      alt="option 1"
      src="option 1"
    />
  </div>,
  <div>
    <img
      alt="option 2"
      src="option 2"
    />
  </div>,
  <div>
    <img
      alt="option 3"
      src="option 3"
    />
  </div>,
  <div>
    <img
      alt="option 4"
      src="option 4"
    />
  </div>,
]
`;
